### PR TITLE
set mode by string

### LINKED ISSUE #431: cant set primary to false via /node endpoint
Primary can only be set to true.

get the error
`[{"fieldNames":["Primary"],"classification":"RequiredError","message":"Required"}]`

this is because the "primary" field is set to required.  But the validation library treats the zero value the same as the value not being defined.

### PR DESCRIPTION
**Pull request by replay**
when using a bool to set a node to primary/secondary we run into the
problem that the `0` value of bool is `false` so when specifying
`false` the parsing library errors saying that a required parameter
has not been passed.
now we're taking the value as a string instead and then parse it into
a bool.
    
fixes #431

### CODE REVIEW

**Comment by woodsaj** [comment]
would it be better to keep use [strconv.ParseBool](https://golang.org/pkg/strconv/#ParseBool) rather then using "primary"/"secondary" text.

```
type NodeStatus struct {
	Primary string `json:"primary" form:"primary" binding:"Required"`
}
```
-----
```
func (s *Server) setNodeStatus(ctx *middleware.Context, status models.NodeStatus) {
        primary, err := strconv.ParseBool(status.Primary)
        if err != nil {
           response.Write(ctx, response.NewError(http.StatusBadRequest, fmt.Sprintf("could parse status to bool. %s", err.Error())))
           return
        }
	cluster.Manager.SetPrimary(primary)
	ctx.PlainText(200, []byte("OK"))
}
```


This would allow us to keep the existing client request `curl http://localhost/node -F primary=true` and not have to update Docs.  Otherwise, the docs need to be updated to refect this change.

**Comment by replay** [comment]
Good idea, that works

File: api/models/node.go [hunk 1]
```diff
@@ -5,7 +5,7 @@ import (
 )
 
 type NodeStatus struct {
-	Primary bool `json:"primary" form:"primary" binding:"Required"`
+	Primary string `json:"mode" form:"mode" binding:"Required"`
```
Comment (Dieterbe) [inline api/models/node.go]: @replay doesn't this change the http parameter from primary to mode? why this change?

Comment (replay) [inline api/models/node.go]: sorry i missed changing that back after i changed them to mode. good that you saw it
